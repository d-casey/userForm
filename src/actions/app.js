import {
  RESET_FORM_STAGE,
  UPDATE_STAGE
} from '../constants/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})
