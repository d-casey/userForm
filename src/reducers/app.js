import {
  UPDATE_STAGE,
  RESET_FORM_STAGE
} from '../constants/actions'

const initialState = {
  stage: 0
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_STAGE:
      return {
        ...state,
        stage: action.payload.stage,
      }

      case RESET_FORM_STAGE:
        return {
          ...state,
          stage: 0
        }

    default:
      return state
  }
}
