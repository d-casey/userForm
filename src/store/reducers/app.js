import {
  UPDATE_STAGE,
  RESET_FORM_STAGE,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

const initialState = {
  stage: 0,
  formCompleted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_STAGE:
      return {
        ...state,
        stage: action.payload.stage,
      }

    case RESET_FORM_STAGE:
      return {
        ...state,
        stage: 0
      }

    case SET_FORM_COMPLETION:
      return {
        ...state,
        formCompleted: true
      }

    case RESET_APP:
      return {
        ...state,
        formCompleted: false,
        stage: 0
      }

    default:
      return state
  }
}
