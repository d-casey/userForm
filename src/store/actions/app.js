import {
  RESET_FORM_STAGE,
  UPDATE_STAGE,
  LOGGING_DETAILS,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

import { resetUserForm, resetPrivacyForm } from '../../store/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})

export const resetAllForms = () => dispatch => {
  dispatch(resetUserForm())
  dispatch(resetPrivacyForm())
}

export const setFormCompletion = () => ({
  type: SET_FORM_COMPLETION
})

export const resetApp = () => ({
  type: RESET_APP
})

export const loggingDetails = () => ({
  type: LOGGING_DETAILS
})

export const logSubmittedFormDetails = (formDetails) => dispatch => {
  dispatch(loggingDetails())
  console.log('formDetails: ', formDetails)
  dispatch(resetAllForms())
  dispatch(setFormCompletion())
}
