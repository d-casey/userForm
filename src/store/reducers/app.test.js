import {
  UPDATE_STAGE,
  RESET_FORM_STAGE,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

import app from './app'

test('it updates the stage correctly', () => {
  expect(app({}, { type: UPDATE_STAGE, payload: { stage: 2 } })).toMatchSnapshot()
})

test('it resets the stage correctly', () => {
  expect(app({}, { type: RESET_FORM_STAGE })).toMatchSnapshot()
})

test('it sets the form to complete correctly', () => {
  expect(app({}, { type: SET_FORM_COMPLETION })).toMatchSnapshot()
})

test('it resets the app correctly', () => {
  expect(app({}, { type: RESET_APP })).toMatchSnapshot()
})


test('it returns the default state when nothing provided', () => {
  expect(app({}, { type: 'UNKNOWN' })).toMatchSnapshot()
})
