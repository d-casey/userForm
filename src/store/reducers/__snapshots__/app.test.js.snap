// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it resets the app correctly 1`] = `
Object {
  "formCompleted": false,
  "stage": 0,
}
`;

exports[`it resets the stage correctly 1`] = `
Object {
  "stage": 0,
}
`;

exports[`it returns the default state when nothing provided 1`] = `Object {}`;

exports[`it sets the form to complete correctly 1`] = `
Object {
  "formCompleted": true,
}
`;

exports[`it updates the stage correctly 1`] = `
Object {
  "stage": 2,
}
`;
